RFS/src/main/java/com/rfs/version_os_2_11/GlobalMetadataCreator_OS_2_11.java

```diff
@@ -131,23 +140,29 @@ private List<String> createTemplates(
                 templatesToCreate.put(templateName, settings);
             }
         } else {
-            // Get the template names
-            List<String> templateKeys = new ArrayList<>();
-            templates.fieldNames().forEachRemaining(templateKeys::add);
-
-            // Create each template
-            for (String templateName : templateKeys) {
+            templates.fieldNames().forEachRemaining(templateName -> {
                 ObjectNode settings = (ObjectNode) templates.get(templateName);
                 templatesToCreate.put(templateName, settings);
-            }
+            });
         }
 
+        return templatesToCreate;
+    }
+
+    private List<String> processTemplateCreation(
+            Map<String, ObjectNode> templatesToCreate,
+            TemplateTypes templateType,
+            MigrationMode mode,
+            IClusterMetadataContext context
+        ) {
+
+        List<String> templateList = new ArrayList<>();
+
         templatesToCreate.forEach((templateName, templateBody) -> {
             log.info("Creating {}: {}", templateType, templateName);
 
             if (mode == MigrationMode.SIMULATE) {
-                var alreadyExists = templateType.alreadyExistsCheck.templateAlreadyExists(client, templateName);
-                if (!alreadyExists) {
```

Comment:
Another rule that IMO makes the code harder to read, but oh well.  We can review the rules in depth later

Reply:
Yeah, I'm not crazy about a fair number of them.  I suspect it's a feature, not a bug, that disabling rules is so hard though.